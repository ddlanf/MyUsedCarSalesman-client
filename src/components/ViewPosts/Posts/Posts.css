   .view-posts-post{
    width: 40%;
    border-radius: 20px;
    margin: 1%;
    margin-top: 6%;
    background-color: rgba(237, 247, 102, 0.952);
    border: #999090 1px solid;
  }

  .view-posts-car-name{
    text-align: center;
    width: 100%;
    font-size: 3vw;
    margin-bottom: 3%;
    font-family: 'Merriweather Sans', sans-serif;
    font-weight: bold;
  }
  
  .view-posts-car-image-box{
    width: 80%;
    margin-left: auto;
    margin-right: auto;
    height: 35%;
  }

  .view-posts-car-image{
    width: 100%;
    border: #999090 1px solid;
  }

  .view-posts-post > ul{
    list-style: none;
    padding: 0;
    text-align: center;
    margin: 0;
    width: 100%;
 }

  .view-posts-lists  li{
    font-family: 'Merriweather Sans', sans-serif;
  }

  ul .view-posts-car-location{
    font-size: 2vw;
    width: 80%;
    margin-left: auto;
    margin-right: auto;
    text-align: left;
    margin-top: -3px;
    margin-bottom: 0;
  }

  ul .view-posts-car-price{
    font-size: 4.5vw;
    margin: 3%;
  }

  ul .view-posts-car-commission{
    font-size: 3.5vw;
    padding-left: 5%;
    padding-right: 5%;
    margin: 1%;
    margin-bottom: 10%;
  }

  .view-post_commision-title{
    display: block;
    font-family: 'Merriweather Sans', sans-serif;
    text-decoration: underline;
  }
  
  .view-posts-lists > li{
      font-size: 3vw;
      margin-bottom: 1%;
  }

  .view-posts-post{
    text-decoration: none;
    color: black;
  }

  .post-link{
    height: auto;
    width: auto;
  }

  @media screen and (min-width: 1000px) {
    .view-posts-post{
      width: 25%;
    }

    .view-posts-post:hover{
      background-color: yellow;
      position: relative;
      bottom: 10px;
      transition: 1s;
    }

    .view-posts-car-name{
      font-size: 30px;
      margin-bottom: 3%;
    }

    .view-posts-lists > li{
      font-size: 30px;
    }

    ul .view-posts-car-price{
      font-size: 40px;
      margin: 0;
    }

    ul .view-posts-car-commission{
      font-size: 30px;
    }

    ul .view-posts-car-location{
      font-size: 20px;
    }
  }