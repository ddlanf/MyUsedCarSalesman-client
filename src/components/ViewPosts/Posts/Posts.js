import React, { Component } from 'react'
import { Link } from 'react-router-dom'
import { dummyData } from '../../dummyData'
import './Posts.css'

export default class Posts extends Component {

    rendorPosts(){
        const { userPosts } = dummyData;
        
        const posts = userPosts.map(post => {
            return (
                        <Link
                            className="view-posts-post"
                            to={`/view-posts/${post.id}`}
                            key={post.id} 
                            >
                            <div 
                                className="view-posts-car-image-box">
                                <img 
                                    className="view-posts-car-image"
                                    src={post.image.src[0]}
                                    alt={post.alt}
                                />
                            </div>
                            <ul
                                className="view-posts-lists">
                                <li>{post.make} {post.model} {post.year}</li>
                                <li>Price:  ${post.price}</li>
                                <li><span className="break">Commission:</span>{post.commission_amount}</li>
                                <li>{post.location}</li>
                            </ul>
                        </Link>
            )
        })

        return posts;
    }

    render() {
        return (
            <>
                {this.rendorPosts()}
            </>
        )
    }
}
