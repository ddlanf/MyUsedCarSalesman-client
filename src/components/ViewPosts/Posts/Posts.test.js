import React from 'react';
import ReactDOM from 'react-dom';
import Posts from './Posts';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom'

describe('Posts component', () => {
    
    it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<BrowserRouter>
                        <Posts
                            posts={[]}
                            images={[]}/>
                    </BrowserRouter>, div);
    ReactDOM.unmountComponentAtNode(div);
    });

    it('renders the UI as expected', () => {
        const tree = renderer
          .create(
                <BrowserRouter>
                    <Posts
                      posts={[]}
                      images={[]}/>
                </BrowserRouter>)
          .toJSON();
        expect(tree).toMatchSnapshot();  
      });
});