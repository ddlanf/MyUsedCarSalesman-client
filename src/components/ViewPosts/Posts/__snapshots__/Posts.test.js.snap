// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Posts component renders the UI as expected 1`] = `null`;
